### PR TITLE
Fix more "incorrect length for compressed encoding" errors

### PR DESCRIPTION
**Pull request by cap10morgan**
Prior to 53362d2ab245b5161a197d44767575ec9af4331d the `recover-public-key-test` added in d083d8168bee92042276d8ebe5a9354163708183 would generate a few errors anytime the recovered key had one or more leading zeroes (exactly like we've seen elsewhere w/ this error).

Adding the code to pad the value to 64 chars before putting the `02` or `03` prefix on fixed it. Now the recovered public key seems to always match the original public key.

### CODE REVIEW

**Comment by cap10morgan** [comment]
Running into some weird issues in CLJS land. I'll ping folks when this is actually ready for review.

**Comment by cap10morgan** [comment]
@fluree/core This is ready for review now. It ended up being a pretty deep rabbit hole, but a fruitful one. I found a few problems in our CLJS code that the secp256k1 tests I had added brought to light. So the additional commits are me fixing those and cleaning some things up.

Biggest takeaway: We need _lots_ more tests in here.

File: src/fluree/crypto/secp256k1.cljc [hunk 1]
```diff
@@ -59,74 +59,65 @@ public key, hex encoded."
...
-                              (instance? BigInteger private) private
-                              (string? private) (BigInteger. ^String private 16))
-                      :cljs (-> (sjcl/bn. private)))]
+  (let [private-bn #?(:clj (cond
+                             (instance? BigInteger private) private
+                             (string? private) (BigInteger. ^String private 16))
+                      :cljs (sjcl/bn. private))]
     (when-not (valid-private? private-bn)
       (throw (ex-info "Invalid private key. Must be big integer and >= 1, <= curve modulus." {:private private})))
     #?(:clj  {:private private-bn
```
Comment (dpetran) [inline src/fluree/crypto/secp256k1.cljc]: Looks like there's an opportunity to push the reader conditional down to the :public keyword here.

**Review by dpetran — APPROVED** [review]
Thanks for adding tests! :1st_place_medal:

**Pull request merged**